### ISSUE TITLE
seedspoints_train

### ISSUE THREAD

**Issue by simi-chen** [BubblyYi/Coronary-Artery-Tracking-via-3D-CNN-Classification#8]
Can I get your trained network? When I was training Seeds_train, my val_loss did not converge all the time, and the effect was very bad during the test
![Total_loss_seeds_model_s0](https://user-images.githubusercontent.com/72775791/95807406-ad340b00-0d3c-11eb-90f7-b44f678be5ea.jpg)

**Comment by lalalala123456789** [comment]
Can I know which Pytorch version do you  use?

**Comment by ziqing-wan99** [comment]
same here

**Comment by ziqing-wan99** [comment]
Hi, 请问你解决了吗？

**Comment by lalalala123456789** [comment]
你们vessel_tree_infer运行成功了吗
















在 2020-10-21 21:12:08，"simi-chen" <notifications@github.com> 写道：

Hi, 请问你解决了吗？

没有

**Comment by ziqing-wan99** [comment]
> 你们vessel_tree_infer运行成功了吗 在 2020-10-21 21:12:08，"simi-chen" <notifications@github.com> 写道： Hi, 请问你解决了吗？ 没有 — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.

没有，我的结果是空的， 太难了

**Comment by BubblyYi** [comment]
Hi, I think it may be because your network is not well trained, the network output some wrong results. It should be noted that you need to write your own code to divide the data sets. Please consider the data distribution and the proportion of positive and negative samples. The key is whether the distribution of the training set and test set is consistent. For example, when training seed point networks, most of the data I generate are negative samples, because there is a small amount of data near blood vessels.

**Comment by simi-chen** [comment]
> 嗨，我认为这可能是因为您的网络训练不足，网络输出了一些错误的结果。应该注意的是，您需要编写自己的代码来划分数据集。请考虑数据分布以及正负样本的比例。关键是训练集和测试集的分布是否一致。例如，在训练种子点网络时，我生成的大多数数据都是负样本，因为在血管附近有少量数据。

您好，很高兴收到您的回复，我想请问一下，您在划分数据集是将8组数据打散划分成训练集，验证集数据的吗，还是采用留一法啊？

**Comment by BubblyYi** [comment]
> > 嗨，我认为这可能是因为您的网络训练不足，网络输出了一些错误的结果。应该注意的是，您需要编写自己的代码来划分数据集。请考虑数据分布以及正负样本的比例。关键是训练集和测试集的分布是否一致。例如，在训练种子点网络时，我生成的大多数数据都是负样本，因为在血管附近有少量数据。
> 
> 您好，很高兴收到您的回复，我想请问一下，您在划分数据集是将8组数据打散划分成训练集，验证集数据的吗，还是采用留一法啊？

It is should be cross-validation

**Comment by chwx08** [comment]
Hi, how to determine the radii of the off-centerline training samples, as they do not lie on the centerline?

**Comment by BubblyYi** [comment]
> Hi, how to determine the radii of the off-centerline training samples, as they do not lie on the centerline?

Good question! When the ball deviates from the centerline, the radius and directions will be recalculated. I use the radius corresponding to the centerline point closest to the current ball coordinate. More important is the direction after offset.

**Comment by chwx08** [comment]
> > Hi, how to determine the radii of the off-centerline training samples, as they do not lie on the centerline?
> 
> Good question! When the ball deviates from the centerline, the radius and directions will be recalculated. I use the radius corresponding to the centerline point closest to the current ball coordinate. More important is the direction after offset.

Thanks for your helpful reply! Recently, I found that in our private dataset, the manually labeled centerlines may sometimes not lie on the true object centerlines. These inaccurate manually labeled may affect the performance of the centerline tracker. Did your datasets have this problem? If yes, how did you solve this?

**Comment by BubblyYi** [comment]
> > > Hi, how to determine the radii of the off-centerline training samples, as they do not lie on the centerline?
> > 
> > 
> > Good question! When the ball deviates from the centerline, the radius and directions will be recalculated. I use the radius corresponding to the centerline point closest to the current ball coordinate. More important is the direction after offset.
> 
> Thanks for your helpful reply! Recently, I found that in our private dataset, the manually labeled centerlines may sometimes not lie on the true object centerlines. These inaccurate manually labeled may affect the performance of the centerline tracker. Did your datasets have this problem? If yes, how did you solve this?

Hi, "_**Standardized evaluation methodology and reference database for evaluating coronary artery centerline extraction algorithms**_" this paper may help you standardize the data annotation process.

**Comment by BubblyYi** [comment]
> > > Hi, how to determine the radii of the off-centerline training samples, as they do not lie on the centerline?
> > 
> > 
> > Good question! When the ball deviates from the centerline, the radius and directions will be recalculated. I use the radius corresponding to the centerline point closest to the current ball coordinate. More important is the direction after offset.
> 
> Thanks for your helpful reply! Recently, I found that in our private dataset, the manually labeled centerlines may sometimes not lie on the true object centerlines. These inaccurate manually labeled may affect the performance of the centerline tracker. Did your datasets have this problem? If yes, how did you solve this?

Moreover, there is a new article (**_Multi-Resolution 3D Convolutional Neural Networks for Automatic Coronary Centerline Extraction in Cardiac CT Angiography Scans_**) recommended to you, hoping to inspire your current research.

**Comment by chwx08** [comment]
> > > > Hi, how to determine the radii of the off-centerline training samples, as they do not lie on the centerline?
> > > 
> > > 
> > > Good question! When the ball deviates from the centerline, the radius and directions will be recalculated. I use the radius corresponding to the centerline point closest to the current ball coordinate. More important is the direction after offset.
> > 
> > 
> > Thanks for your helpful reply! Recently, I found that in our private dataset, the manually labeled centerlines may sometimes not lie on the true object centerlines. These inaccurate manually labeled may affect the performance of the centerline tracker. Did your datasets have this problem? If yes, how did you solve this?
> 
> Moreover, there is a new article (**_Multi-Resolution 3D Convolutional Neural Networks for Automatic Coronary Centerline Extraction in Cardiac CT Angiography Scans_**) recommended to you, hoping to inspire your current research.

Many thanks for your sharings! I think these will be helpful to our research!

**Comment by dickysg** [comment]
I am just wondering what is the usage of seedpoint in this context? are they randomize point place in the image to identify the coronary artery localize position from the 3D image?

**Comment by Ritaxuu** [comment]
> > 你们vessel_tree_infer运行成功了吗 在 2020-10-21 21:12:08，"simi-chen" [notifications@github.com](mailto:notifications@github.com) 写道： Hi, 请问你解决了吗？ 没有 — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.
> 
> 没有，我的结果是空的， 太难了

结果空的，跑不出。。。太难了

**Comment by zyx3216825** [comment]
谁能提供一下cta08数据呢，官网注册邮件也没有回应

**Comment by zyx3216825** [comment]
> Can I get your trained network? When I was training Seeds_train, my val_loss did not converge all the time, and the effect was very bad during the test
> ![Total_loss_seeds_model_s0](https://user-images.githubusercontent.com/72775791/95807406-ad340b00-0d3c-11eb-90f7-b44f678be5ea.jpg)

你这个问题解决了吗，我也遇到和你一样的问题，验证loss不拟合，我是前7套训练，最后一套验证

**Comment by AManHasNoName12138** [comment]
> > Can I get your trained network? When I was training Seeds_train, my val_loss did not converge all the time, and the effect was very bad during the test
> > ![Total_loss_seeds_model_s0](https://user-images.githubusercontent.com/72775791/95807406-ad340b00-0d3c-11eb-90f7-b44f678be5ea.jpg)
> 
> 你这个问题解决了吗，我也遇到和你一样的问题，验证loss不拟合，我是前7套训练，最后一套验证

跑个题，请问一下这一行代码需要的.csv文件是什么时候创建的呢，我跑train的时候报错不存在改文件，谢谢[https://github.com/BubblyYi/Coronary-Artery-Tracking-via-3D-CNN-Classification/blob/57ecdf631884946d8cb979a63932dd5f07d108d1/centerline_train_tools/centerline_train_tools.py#L16](url)

**Comment by AManHasNoName12138** [comment]
https://github.com/BubblyYi/Coronary-Artery-Tracking-via-3D-CNN-Classification/blob/57ecdf631884946d8cb979a63932dd5f07d108d1/centerline_train_tools/centerline_train_tools.py#L16

**Comment by renshengruxge** [comment]
> > Can I get your trained network? When I was training Seeds_train, my val_loss did not converge all the time, and the effect was very bad during the test
> > ![Total_loss_seeds_model_s0](https://user-images.githubusercontent.com/72775791/95807406-ad340b00-0d3c-11eb-90f7-b44f678be5ea.jpg)
> 
> 你这个问题解决了吗，我也遇到和你一样的问题，验证loss不拟合，我是前7套训练，最后一套验证



> > Can I get your trained network? When I was training Seeds_train, my val_loss did not converge all the time, and the effect was very bad during the test
> > ![Total_loss_seeds_model_s0](https://user-images.githubusercontent.com/72775791/95807406-ad340b00-0d3c-11eb-90f7-b44f678be5ea.jpg)
> 
> 你这个问题解决了吗，我也遇到和你一样的问题，验证loss不拟合，我是前7套训练，最后一套验证

你好，请问你是否方便提供数据集，官网最近登录不进去，现在没有办法下载对应的数据集，如果方便的话数据可以发送到我的邮箱2823212088@qq.com，谢谢！

**Comment by renshengruxge** [comment]
> > > 你们vessel_tree_infer运行成功了吗 在 2020-10-21 21:12:08，"simi-chen" [notifications@github.com](mailto:notifications@github.com) 写道： Hi, 请问你解决了吗？ 没有 — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.
> > 
> > 
> > 没有，我的结果是空的， 太难了
> 
> 结果空的，跑不出。。。太难了

你好，请问你是否方便提供数据集，官网最近登录不进去，现在没有办法下载对应的数据集，如果方便的话可以发送到我的邮箱2823212088@qq.com，谢谢！

**Comment by M2Dlm** [comment]
> > > > 你们vessel_tree_infer运行成功了吗 在 2020-10-21 21:12:08，"simi-chen" [notifications@github.com](mailto:notifications@github.com) 写道： Hi, 请问你解决了吗？ 没有 — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.
> > > 
> > > 
> > > 没有，我的结果是空的， 太难了
> > 
> > 
> > 结果空的，跑不出。。。太难了
> 
> 你好，请问你是否方便提供数据集，官网最近登录不进去，现在没有办法下载对应的数据集，如果方便的话可以发送到我的邮箱[2823212088@qq.com](mailto:2823212088@qq.com)，谢谢！




> > > > 你们vessel_tree_infer运行成功了吗 在 2020-10-21 21:12:08，"simi-chen" [notifications@github.com](mailto:notifications@github.com) 写道： Hi, 请问你解决了吗？ 没有 — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.
> > > 
> > > 
> > > 没有，我的结果是空的， 太难了
> > 
> > 
> > 结果空的，跑不出。。。太难了
> 
> 你好，请问你是否方便提供数据集，官网最近登录不进去，现在没有办法下载对应的数据集，如果方便的话可以发送到我的邮箱[2823212088@qq.com](mailto:2823212088@qq.com)，谢谢！

你好，数据集搞定了吗，如果可以的话，能否发送相关数据或链接到我的邮箱（996374875@qq.com），谢谢！

**Comment by chuxiang93** [comment]
> 你好，数据集搞定了吗，如果可以的话，能否发送相关数据或链接到我的邮箱（[996374875@qq.com](mailto:996374875@qq.com)），谢谢！

你好，数据集搞定了吗，如果可以的话，能否发送相关数据或链接到我的邮箱（[154578868@qq.com](mailto:154578868@qq.com)），谢谢！

**Comment by XuyangPan** [comment]
> > Can I get your trained network? When I was training Seeds_train, my val_loss did not converge all the time, and the effect was very bad during the test
> > ![Total_loss_seeds_model_s0](https://user-images.githubusercontent.com/72775791/95807406-ad340b00-0d3c-11eb-90f7-b44f678be5ea.jpg)
> 
> 你这个问题解决了吗，我也遇到和你一样的问题，验证loss不拟合，我是前7套训练，最后一套验证

你好，请问你是否方便提供数据集，官网最近登录不进去没有办法下载数据集，如果方便的话请发送到我的邮箱[xuyangpan@yeah.net](mailto:xuyangpan@yeah.net)，谢谢！

**Comment by XuyangPan** [comment]
> > 你好，数据集搞定了吗，如果可以的话，能否发送相关数据或链接到我的邮箱（[996374875@qq.com](mailto:996374875@qq.com)），谢谢！
> 
> 你好，数据集搞定了吗，如果可以的话，能否发送相关数据或链接到我的邮箱（[154578868@qq.com](mailto:154578868@qq.com)），谢谢！

你好，数据集搞定了吗，如果可以的话，能否发送相关数据或链接到我的邮箱（[xuyangpan@yeah.net](mailto:xuyangpan@yeah.net)），谢谢！

**Comment by Manasi-25** [comment]
> > Hi, how to determine the radii of the off-centerline training samples, as they do not lie on the centerline?
> 
> Good question! When the ball deviates from the centerline, the radius and directions will be recalculated. I use the radius corresponding to the centerline point closest to the current ball coordinate. More important is the direction after offset.

Could you please give me access to the dataset
